### ISSUE TITLE
DVPackager Windows

### ISSUE THREAD

**Issue by AoifeFitz** [mipops/dvrescue#299]
Hello all, 

Great to see this project up and running! Looks like it will be an invaluable resource for DV capture, QC and preservation.

Please excuse any incorrect terminology I am about to use - I am not very fluent in git-speak, cloning etc.. 

I have a .dv file that goes out of sync when transcoded in any way, I am wondering if it is the same issue that Dave Rice flagged in the AMIA 2020 conference where blank audio frames will be filled in with the subsequent audio data by ffmpeg when transcoding, without heed being paid to the video frames they should correspond to.

I have downloaded the latest Windows build from here (https://mediaarea.net/download/snapshots/binary/dvrescue-gui/20210415-2/)

I can run the GUI no problem, I am wondering if at the moment there is an option to export individual files based on recording breaks from the Package section in the GUI? I know you are all busy working on it, so maybe this feature won't be available for a while from the GUI.

Or could I go about doing so using the command line? When I cd into the downloaded folder, and try to run dvpackager, I get the attached output.
![dvpackager_error](https://user-images.githubusercontent.com/71381132/115247805-04b59800-a11f-11eb-82ee-01680f1f1aa6.JPG)

I feel this is my own inexperience in compiling packages to run in Command Prompt, so apologies as this may not be an issue, more so me hoping for advice. I am using Windows 10.

Thanks very much for reading and looking forward to seeing DVRescue develop!

Aoife

**Comment by kieranjol** [comment]
> 
> Or could I go about doing so using the command line? When I cd into the downloaded folder, and try to run dvpackager, I get the attached output.
> ![dvpackager_error](https://user-images.githubusercontent.com/71381132/115247805-04b59800-a11f-11eb-82ee-01680f1f1aa6.JPG)
> 
> I feel this is my own inexperience in compiling packages to run in Command Prompt, so apologies as this may not be an issue, more so me hoping for advice. I am using Windows 10.

Hi Aoife - if memory serves, those tools like dvpackager in there are actually bash scripts which will not run natively in a windows CMD prompt. There's a few ways around this, but if you're up for the challenge, I was able to get it working in windows in the Ubuntu subsystem - https://docs.microsoft.com/en-us/windows/wsl/install-win10
That should allow you to use the bash tools on the command line in windows. I was generally able to get all of the tools working correctly, though navigating around and actually finding your DV file can be annoying. Hopefully there's a way to get all this done in the GUI, I'm just a bit out of the loop on that.
Looks like @elderorb is working on cygwin integration to make all this happen a lot smoother in the actual GUI in windows though.

**Comment by dericed** [comment]
It's true, https://github.com/mipops/dvrescue/pull/288 should bring dvpackager in the ui to Windows. We'll send a link when it's ready to test.

**Comment by AoifeFitz** [comment]
Hey @kieranjol ! Thanks for the tip, I will check out this link and also brush up my memory of the various scripts! And thank you @dericed , I'm looking forward to testing it out :)

**Comment by ElderOrb** [comment]
First POC on cygwin integration seems to be working: https://github.com/mipops/dvrescue/pull/288

**Comment by dericed** [comment]
Hi @AoifeFitz @privatezero, there's now a version of the DVRescue GUI with the packaging process at https://mediaarea.net/download/snapshots/binary/dvrescue-gui/20210420/. The GUI on the package tab should let you open a file and set the following options and run the packaging process:

```
 -s       (split the output file at recording start markers)
 -d       (split the output file at non-consecutive recording timestamps)
 -t       (split the output file at non-consecutive timecode values)
 -S       (enables technical subtitle track to show timecode,
           recording time, and dv error data and a caption track if
           there are captions to represent in the source DV)
```

Currently the GUI won't show any feedback from running it (besides the debug log) but next to the input should be an output directory with the packaged result. Please let me know if this works for you.

**Comment by AoifeFitz** [comment]
Thanks so much for sending me this. I downloaded and ran the file through the GUI packager, testing each of the options. Each option bar -S outputted an XML file, I have attached it here as a .txt file. Is this to be expected, or should the output be dv files split into segments? I may be wrong with my analysis of the file, but it plays beautifully as .DV, the sync issues arise when it is transcoded. I am guessing that the issues arise from tape overwrites etc.
[d9eda6f8-0ea0-48fe-bdae-0575588673fa.dv.dvrescue.txt](https://github.com/mipops/dvrescue/files/6349412/d9eda6f8-0ea0-48fe-bdae-0575588673fa.dv.dvrescue.txt)

**Comment by dericed** [comment]
Hi @AoifeFitz, could you please update to the latest dvrescue command line tools at https://mediaarea.net/download/snapshots/binary/dvrescue/ and the latest gui at https://mediaarea.net/download/snapshots/binary/dvrescue-gui/ and then re-run this test. Also please confirm that you have ffmpeg 4.4 or later installed.

Then when running the packager part of the GUI, you should see a packaged file appear, if not, please open the debug window and paste the output over here. Thanks for testing :)

**Comment by AoifeFitz** [comment]
Hi @dericed , I think I have updated everything accordingly (disclaimer that my compilation skills are not the best) and upon running DVPackager through the GUI with the -d flag checked, a sidecar file is generated with a .xml file present. I will attach a copy below if you would like to take a look. 
[d9eda6f8-0ea0-48fe-bdae-0575588673fa.dv.dvrescue.txt](https://github.com/mipops/dvrescue/files/6351362/d9eda6f8-0ea0-48fe-bdae-0575588673fa.dv.dvrescue.txt)


I also may be missing a step in the compilation, flagged by the image I have attached. Would this impact the GUI?
![compiling_problem](https://user-images.githubusercontent.com/71381132/115565877-acfb6600-a2b1-11eb-9c87-cd39249aba96.JPG)


Happy to be testing! Only hope I am testing correctly :) 

Here is the output of the debug window:

beginning commands log
C:/Users/yhuang/Desktop/dvrescue_GUI_0.20.11.20210421_Windows_x64/cygwin/bin/bash.exe C:/Users/yhuang/Desktop/dvrescue_GUI_0.20.11.20210421_Windows_x64/tools/dvpackager -v -e mov -X /cygdrive/c/Users/yhuang/Desktop/dvrescue_GUI_0.20.11.20210421_Windows_x64/scripts/xml.sh
 -F /cygdrive/c/Users/yhuang/Desktop/dvrescue_GUI_0.20.11.20210421_Windows_x64/scripts/ffmpeg.sh
 -D /cygdrive/c/Users/yhuang/Desktop/dvrescue_GUI_0.20.11.20210421_Windows_x64/scripts/dvrescue.sh
 -M /cygdrive/c/Users/yhuang/Desktop/dvrescue_GUI_0.20.11.20210421_Windows_x64/scripts/xml.sh
 -d /cygdrive/E/ballymun/Aoife_test_new/d9eda6f8-0ea0-48fe-bdae-0575588673fa.dv


[31mFFmpeg's libavformat version  58.65.101 or greater is recommend and /cygdrive/c/mas/media-autobuild_suite-master/local64/bin-video/ffmpeg is running .(B[m




[31mPlease update ffmpeg or consider installing amiaopensource/amiaos/ffmpegdecklink.(B[m




[31mUsing this current install of ffmpeg may result in out of sync audio.(B[m




Packaging d9eda6f8-0ea0-48fe-bdae-0575588673fa.dv, making dvrescue outputs




[35m#No xmlstarlet error(B[m




[35m#Trying to make a directory at /cygdrive/E/ballymun/Aoife_test_new/d9eda6f8-0ea0-48fe-bdae-0575588673fa.dv_dvrescue(B[m




[35m#Trying to move the temporary xml from /tmp/dvpackager.kWcsnS to /cygdrive/E/ballymun/Aoife_test_new/d9eda6f8-0ea0-48fe-bdae-0575588673fa.dv_dvrescue/d9eda6f8-0ea0-48fe-bdae-0575588673fa.dv.dvrescue.xml(B[m




[35m#DV_FRAME_COUNT=()(B[m




[35m#SOURCE_FORMAT=()(B[m




[35m#Trying to present a table of ranges(B[m





#         St='Flagged Start of a recording', ncTC='non-continuous timecode value', ncR='non-continuous recording timestamp value', 1st='first frame is sequence'
| PTS Range                         | Duration | Frame Range     | Timecode    | Recording Timestamp | Size      | Frame Rate | DAR   | ChSub | Audio     | St | ncTC | ncR | 1st |




|                 -                 |    0.000 |      0 -      0 |             |                     |           |            |       |       |  ch     0 |    |      |     |     |









[32mThe results will be written to /cygdrive/E/ballymun/Aoife_test_new/d9eda6f8-0ea0-48fe-bdae-0575588673fa.dv_dvrescue(B[m




[31mERROR: d9eda6f8-0ea0-48fe-bdae-0575588673fa.dv contained  DV frames, but the outputs contain 0 frames of DV.(B[m

**Comment by dericed** [comment]
Hi @AoifeFitz , the xml you attached at https://github.com/mipops/dvrescue/files/6351362/d9eda6f8-0ea0-48fe-bdae-0575588673fa.dv.dvrescue.txt is from dvrescue 0.20.06 which likely is too old to work here. I see the path to your dvrescue cli in the image you pasted, but I'm surprised that updating to the latest at https://mediaarea.net/download/snapshots/binary/dvrescue/ didn't replace that version of dvrescue.exe.

**Comment by AoifeFitz** [comment]
Yes, I see yet I seem to be working from 0.20.11 according to command prompt - I'll have another go at compiling and updating and update here when done :)
